Add admin events for realm create/delete

tsaarni opened the pull request:
This PR adds realm `CREATE` and `DELETE` events at realm creation and deletion for having audit trail also for those operations and for helping troubleshooting.

Fixes #10733

The event log listeners will of course receive these events normally, but when saving events to database there is additional complication. Stored admin events are modeled as resources under realms. They are fetched by `GET /{realm}/admin-events`. That works for fetching events about operations for "child" resources such as users of a realm but it raises a question: which realm would be used to fetch events of realm creation and deletion?

Note: Proposal 1 is currently implemented by this PR.

**Proposal 1:** 

Associate events with the realm of authenticated administrator instead of the realm that is being created or deleted. Add the realm name as resource path in the event. In this way, `master` realm becomes the "parent" of all realms when it comes to fetching the create/delete realm events. There is asymmetry in this proposal, since the existing `UPDATE` events are of course saved to the realm being updated.

**Proposal 2:**

Associate events with the realm being created/deleted. The resource path of the event becomes empty since the operation addresses realm itself and not a resource under realm. 

In proposal 2, it must be accepted that event storing will not work reliably because:

* `CREATE` can be lost since at realm creation the realm does not exist yet and saving is not likely to be enabled (at least when creating from GUI without import file).
* `DELETE` is lost at the same time the realm delete operation is executed.

Signed-off-by: Tero Saarni <tero.saarni@est.tech>

sreekesh93 commented:
any updates on this PR ? Is there any issue on producing event at time or realm creation of deltion?

sreekesh93 commented:
@stianst It would be really helpful if you can give some suggestion for this PR

tsaarni commented:
It would be great to have feedback on this feature! Thank you!

tsaarni committed:
Add admin events for realm create/delete.

Signed-off-by: Tero Saarni <tero.saarni@est.tech>

mposolda commented on testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/event/AdminEventTest.java:
```diff
@@ -21,10 +21,7 @@
-import org.keycloak.representations.idm.AuthDetailsRepresentation;
-import org.keycloak.representations.idm.RealmRepresentation;
-import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.representations.idm.*;
```
Can you please remove star imports?

mposolda reviewed (changes requested):
LGTM, Thanks for the PR!

Can you please rebase to latest main? Also added one inner inline comment

tsaarni commented:
> LGTM, Thanks for the PR!
> 
> Can you please rebase to latest main? Also added one inner inline comment

@mposolda Thank you for the review! 

I've rebased with main and now CI looks clean again, thanks to https://github.com/keycloak/keycloak/pull/17465.

mposolda reviewed (approved):
@tsaarni Thanks! Hope we're able to leverage more of your awesome contributions...
